src/operations/blas3/gemm_local_tensorcore.hpp

```diff
@@ -0,0 +1,814 @@
+ *               memory consumption) - may be useful in combination with TranA
+ * @tparam NbcA  iff true, avoids bank conflicts when accessing blocks of
+ *               matrix B in local memory (slightly increases local
+ *               memory consumption) - may be useful in combination with TranB
+ * @tparam ClSize  the size of the cache line of the architecture
+ *                 (If the value passed in is smaller than the actual cache
+ *                 line size, some values fetched will be wasted, which can
+ *                 significantly reduce performance. It can be set to a
+ *                 multiple of the physical cache line size. In this case, it
+ *                 will significantly increase local memory usage, but
+ *                 will result in fewer local barriers.)
+ * @tparam TileType  determines the size of the local, work group, and top
+ *                   level tiles to use, see Tile
+ * @tparam TransA  iff true, matrix A will be transposed on the fly
+ * @tparam TransB  iff true, matrix B will be transposed on the fly
+ * @tparam element_t  type of matrix elements
+ * @tparam is_beta_zero True if beta == 0.
+ * @tparam VectorSize The packet size to be used for vectorization.
+ * @tparam batch_type the type of batch strideded /interleaved
+ * @tparam UseTensorcores boolean parameter to decide whether to use Tensorcores
+ * or not
+ */
+#ifdef SB_ENABLE_JOINT_MATRIX
+template <typename input_t, typename output_t, bool DoubleBuffer, bool NbcA,
+          bool NbcB, int ClSize, typename TileType, bool TransA, bool TransB,
+          typename element_t, bool is_beta_zero, int VectorSize>
+class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, TileType,
+           TransA, TransB, element_t, is_beta_zero,
+           static_cast<int>(gemm_memory_t::local),
+           static_cast<int>(gemm_algorithm_t::standard),
+           static_cast<int>(gemm_vectorization_t::full), VectorSize,
+           static_cast<int>(gemm_batch_type_t::strided), true> {
+ public:
+  using tile_type = TileType;
+  using value_t = element_t;
+  using index_t = typename std::make_signed<typename input_t::index_t>::type;
+  using packetize_t = Packetize<VectorSize, value_t, index_t>;
+  using vector_t = typename packetize_t::PacketType;
+  using address_t = cl::sycl::access::address_space;
+
+  // enable easier access to tile dimensions
+  static constexpr index_t item_rows = tile_type::item_rows;
+  static constexpr index_t item_cols = tile_type::item_cols;
+  static constexpr index_t wg_rows = tile_type::wg_rows;
+  static constexpr index_t wg_cols = tile_type::wg_cols;
+  static constexpr index_t sg_rows = tile_type::sg_rows;
+  static constexpr index_t sg_cols = tile_type::sg_cols;
+  static constexpr index_t tl_rows = tile_type::tl_rows;
+  static constexpr index_t tl_cols = tile_type::tl_cols;
+  static constexpr index_t tile_size = tl_rows * tl_cols;
+  static constexpr index_t sg_size = 32;
```

Comment:
Should this be computed from sg_rows and sg_cols? If it has to be 32, perhaps we should then have a static assert.